src/fromager/sources.py

```diff
@@ -59,6 +59,13 @@ def download_source(
         ) as err:
             logger.debug(f"{req.name}: failed to resolve {req} using {url}: {err}")
             continue
+
+        # Validate version string by passing it to Version
+        try:
+            Version(str(version))
```

Comment:
I think `version` is already a string so `str()` shouldn't be needed

Reply:
I checked, its not a `str()`. The e2e tests were failing because of that

Reply:
You are right. In fact `default_download_source` is returning a version object and is already doing this validation then. It gets the version from: https://github.com/python-wheel-build/fromager/blob/main/src/fromager/sources.py#L138

and in `resolve_dist`, the `candidate.version` field is being returned which is already a `Version`: https://github.com/python-wheel-build/fromager/blob/main/src/fromager/resolver.py#L44

We might want to fix this mismatch

Reply:
Hmm let me check. Does it mean we might not need validation again at all?

Reply:
The user defined override will a return a string. That's the problem

Reply:
Okay, got it!